zeppelin-web/src/app/notebook/notebook.controller.js

```diff
@@ -1069,8 +1064,55 @@ function NotebookCtrl ($scope, $route, $routeParams, $location, $rootScope,
     $scope.note.config.personalizedMode = isPersonalized
   })
 
+  $scope.$on('$routeChangeStart', function (event, next, current) {
+    if ($scope.note && $scope.note.paragraphs) {
```

Comment:
MINOR: It's better to return earlier. For example, 

`if (!$scope.note || !$scope.note.paragraphs) { return }`